cli/install/install.go

```diff
@@ -0,0 +1,572 @@
+}
+
+// Packet is a struct containing sys and install name of the packet.
+type Packet struct {
+	sysName     string
+	installName string
+}
+
+type DistroInfo struct {
+	Name         string
+	Vendor       string
+	Version      string
+	Architecture string
+}
+
+var (
+	PrettyName = regexp.MustCompile(`^PRETTY_NAME=(.*)$`)
+	ID         = regexp.MustCompile(`^ID=(.*)$`)
+	VersionID  = regexp.MustCompile(`^VERSION_ID=(.*)$`)
+)
+
+func GetDistroName() *DistroInfo {
+	var distroInfo DistroInfo
+	if _, err := os.Stat("/lib64/ld-linux-x86-64.so.2"); err == nil {
+		distroInfo.Architecture = "amd64"
+	} else if _, err := os.Stat("/lib/ld-linux.so.2"); err == nil {
+		distroInfo.Architecture = "i386"
+	}
+
+	distoFile, err := os.Open("/etc/os-release")
+	if err != nil {
+		return nil
+	}
+	defer distoFile.Close()
+
+	scanner := bufio.NewScanner(distoFile)
+	for scanner.Scan() {
+		if m := PrettyName.FindStringSubmatch(scanner.Text()); m != nil {
+			distroInfo.Name = strings.Trim(m[1], `"`)
+		} else if m := ID.FindStringSubmatch(scanner.Text()); m != nil {
+			distroInfo.Vendor = strings.Trim(m[1], `"`)
+		} else if m := VersionID.FindStringSubmatch(scanner.Text()); m != nil {
+			distroInfo.Version = strings.Trim(m[1], `"`)
+		}
+	}
+	return &distroInfo
+}
+
+func DetectOsName() string {
+	if runtime.GOOS == "darwin" {
+		return "darwin"
+	}
+	if runtime.GOOS == "windows" {
+		return "windows"
+	}
+	if runtime.GOOS == "linux" {
+		distroInfo := GetDistroName()
+		return distroInfo.Name
+	}
+	return ""
+}
+
+// isDeprecated checks wethere version of programm is below 1.10.0.
+func isDeprecated(version string) bool {
+	splitedVersion := strings.Split(version, ".")
+	if len(splitedVersion) < 2 {
+		return false
+	}
+	if splitedVersion[0] == "1" && len(splitedVersion[1]) < 2 {
+		return true
+	}
+	return false
+}
+
+// getLatestTarantoolTag returns latest tag from tarantool repository.
+func getLatestTarantoolTag() (string, error) {
+	cmd := exec.Command("git", "ls-remote", "--tags", "--sort="+
+		"v:refname", "https://github.com/tarantool/tarantool.git")
+	readPipe, writePipe, _ := os.Pipe()
+	cmd.Stdout = writePipe
+	cmd.Stderr = os.Stderr
+	cmd.Start()
+	err := cmd.Wait()
+	writePipe.Close()
+	if err != nil {
+		return "", err
+	}
+
+	var buf bytes.Buffer
+	io.Copy(&buf, readPipe)
+	versions := buf.String()
+	versionsArray := strings.Split(versions, "\n")
+	latestTag := versionsArray[len(versionsArray)-2]
+	latestTag = strings.TrimSuffix(latestTag, "^{}")
+	trimpos := strings.LastIndex(latestTag, "/") + 1
+	latestTag = latestTag[trimpos:]
+	return latestTag, nil
+}
+
+// copy copies file from source to destination.
+func copy(src string, dst string) error {
+	// Read all content of src to data.
+	data, err := ioutil.ReadFile(src)
+	if err != nil {
+		return err
+	}
+	// Write data to dst.
+	err = ioutil.WriteFile(dst, data, 0777)
+	return err
+}
+
+func copyDir(src string, dst string) error {
+	cmd := exec.Command("cp", "--recursive", src, dst)
```

Comment:
Better write callback for filepath.Walk function.

Reply:
Changed